tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ClassEmitter.scala

```diff
@@ -70,40 +70,71 @@ private[emitter] final class ClassEmitter(jsGen: JSGen) {
       }
 
       if (!tree.kind.isJSClass) {
+        assert(tree.jsSuperClass.isEmpty, className)
         entireClassDefWithGlobals
       } else {
         // Wrap the entire class def in an accessor function
         import TreeDSL._
         implicit val pos = tree.pos
 
-        val createClassValueVar =
-          envFieldDef("b", className, js.Undefined(), mutable = true)
+        val genStoreJSSuperClass = tree.jsSuperClass.map { jsSuperClass =>
+          for (rhs <- desugarExpr(jsSuperClass, resultType = AnyType)) yield {
+            js.VarDef(genStoredSuperClass().ident, Some(rhs))
```

Comment:
Not sure `genStoredSuperClass` deserves to be special. Just use the `envField` methods?

Reply:
Like you said in the other comment, it's indeed not an abuse of the mechanism (just of the name). So indeed it doesn't deserve to be special.